spec/factories/user_factory.rb

```diff
@@ -58,9 +58,10 @@
       user.pref.save if factory.preferences.present?
 
       if user.notification_settings.empty?
-        all_true = NotificationSetting.all_settings.index_with(true)
+        date_settings = [NotificationSetting::START_DATE, NotificationSetting::DUE_DATE, NotificationSetting::OVERDUE]
```

Comment:
We could do the `index_with` in this line, as it's more consistent with the `date_settings` initialization.